lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2374,9 +2453,48 @@ bool LLVMToSPIRV::transExecutionMode() {
       }
     }
   }
+
+  transFPContract();
+
   return true;
 }
 
+void LLVMToSPIRV::transFPContract() {
+  for (Function &F : *M) {
+    SPIRVValue *TranslatedF = getTranslatedValue(&F);
+    if (!TranslatedF) {
+      continue;
+    }
```

Comment:
I would assert that all functions were translated, otherwise, how can you be sure that you have gathered all information required for fp-contract handling?

Moreover, this function is called almost at the end of [`translate`](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/521/commits/8504e880a16f5d8794662aedac23ab90175d9a72#diff-3d6ba7d110dbca252605cea8fff24de7R2312)

Reply:
Intrinsics and SPIR-V built-in functions are not translated to SPIR-V functions, so they are excluded by this check.

> how can you be sure that you have gathered all information required for fp-contract handling?

Everything that gets to `transFunction` is handled. Here in `transFPContract` we only apply fp-contract mode based on the information we gathered in all `transFunction` calls.